packages/gatsby/src/schema/index.js

```diff
@@ -124,7 +127,10 @@ const build = async ({ parentSpan, fullMetadataBuild = true }) => {
     type: `SET_SCHEMA`,
     payload: schema,
   })
-
+  fs.writeFileSync(
```

Comment:
https://www.gatsbyjs.org/packages/gatsby-plugin-schema-snapshot/ does this too? Should we duplicate logic or converge and make it a default plugin?

Reply:
I tried that plugin, and the version of the schema provided was woefully incomplete. no scalars, directives, and many of the plugin types were missing

Reply:
Yup! Is there any value in updating that plugin and letting all of this logic live there? We could always bump its major and publish a version that includes all of this.

Mostly wondering because this will make much of that plugin irrelevant?

Reply:
my idea was to see what changes we need to make to core to provide any necessary interfaces a plugin might need to do these things. it would replace the need for this plugin, though.

Reply:
since it will soon be used by the typescript tooling that will be shipped with gatsby projects by default, it may or may not make sense to provide this via a plugin

Reply:
Yeah, that seems reasonable.

Reply:
so, to summarize/confirm @sidharthachatterjee - we will not worry about this existing plugin since this behavior will ship with gatsby by default soon?

when I attempted to add the plugin interface to generate a more full representation of the schema than that other plugin, I found it required a lot of effort for something that would be shipping with gatsby by default anyways